e2e-tests/cypress/e2e/users.cy.js

```diff
@@ -0,0 +1,52 @@
+describe('Users page', () => {
+    before(() => {
+        cy.login();
+    });
+
+    beforeEach(() => {
+        cy.visit('/compute_plans');
+        cy.getDataCy('menu-button').click();
+        cy.get('[data-user-role]')
+            .invoke('data', 'user-role')
+            .then((userRole) => {
+                if (userRole === 'ADMIN') {
+                    cy.visit('/users');
+                }
+            });
+    });
+
+    it('can create user', () => {
+        cy.getDataCy('create-user').click();
+        cy.getDataCy('username-input').type('Test');
+        cy.getDataCy('password-input').type('Azertyuiop123456789$');
+        cy.getDataCy('submit-form').click();
+
+        cy.get('[data-name="Test"]').first().should('exist');
+    });
+
+    it('can update user', () => {
+        cy.get('[data-name="Test"]')
+            .first()
+            .should('exist')
+            .then(($el) => {
+                cy.wrap($el).should('have.data', 'role', 'USER');
+                cy.wrap($el).click();
+            });
+        cy.get('select').eq(0).select('ADMIN');
+        cy.getDataCy('submit-form').click();
+
+        cy.wait(500);
+        cy.get('[data-name="Test"]')
+            .first()
+            .then(($el) => {
```

Comment:
I've just seen that the `should` command has built-in **retry-ability**. Cypress will automatically re-run the assertion until it passes or the timeout is reached (compared to `then` which doesn't have retry-ability, meaning it will not automatically re-run if the assertion within it fails).

By replacing `then` with `should`, we're telling Cypress to keep checking the condition (in this case, whether the user role has been updated to 'ADMIN') until it's true or until the timeout is reached.

Note that if the timeout is too short, you can increase it `cy.get('[data-name="Test"]', { timeout: 500 })`

```suggestion
            cy.get('[data-name="Test"]')
        .first()
        .should(($el) => {
            expect($el).to.have.data('role', 'ADMIN');
        });
```

Reply:
Cleaner indeed :+1: I'll try this !

Reply:
I thought the `should()` i was already using would do this inside the `then`